tests/components/application-connector/test/compass-runtime-agent/testkit/compassruntimeagentinit/compassconnection.go

```diff
@@ -0,0 +1,71 @@
+package compassruntimeagentinit
+
+import (
+	"context"
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/kyma/components/compass-runtime-agent/pkg/apis/compass/v1alpha1"
+	"github.com/kyma-project/kyma/tests/components/application-connector/test/compass-runtime-agent/testkit/compassruntimeagentinit/types"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type compassConnectionCRConfiguration struct {
+	compassConnectionInterface CompassConnectionInterface
+}
+
+//go:generate mockery --name=CompassConnectionInterface
+type CompassConnectionInterface interface {
+	Get(ctx context.Context, name string, opts v1.GetOptions) (*v1alpha1.CompassConnection, error)
+	Create(ctx context.Context, compassConnection *v1alpha1.CompassConnection, opts v1.CreateOptions) (*v1alpha1.CompassConnection, error)
+	Update(ctx context.Context, compassConnection *v1alpha1.CompassConnection, opts v1.UpdateOptions) (*v1alpha1.CompassConnection, error)
+	Delete(ctx context.Context, name string, opts v1.DeleteOptions) error
+}
+
+func NewCompassConnectionCRConfiguration(compassConnectionInterface CompassConnectionInterface) compassConnectionCRConfiguration {
+	return compassConnectionCRConfiguration{
+		compassConnectionInterface: compassConnectionInterface,
+	}
+}
+
+func (cc compassConnectionCRConfiguration) Do() (types.RollbackFunc, error) {
+
+	compassConnectionCR, err := cc.compassConnectionInterface.Get(context.TODO(), "compass-connection", meta.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = cc.compassConnectionInterface.Delete(context.TODO(), "compass-connection", meta.DeleteOptions{})
+	if err != nil {
+		return nil, err
+	}
+	compassConnectionCR.ResourceVersion = ""
+
+	compassConnectionCRBackup := compassConnectionCR.DeepCopy()
+	compassConnectionCRBackup.ObjectMeta.Name = "compass-connection-backup"
+	_, err = cc.compassConnectionInterface.Create(context.TODO(), compassConnectionCRBackup, meta.CreateOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
Just asking if I understand it correctly. We already deleted the `compass-connection` resource in this place but there is no rollback function because it'd be recreated by the Agent, is it right?

Reply:
It is supposed to works as follows:
- in `Do` we delete original CompassConnection CR so that the CR will be recreated by the Compass Runtime Agent. Additionally we create a backup version of the CR.
- In rollback function backup CR is deleted ind CompassConnection CR restored

I think it needs some testing to make sure it is a correct approach. I didn't checked the case when Compass Runtime Agent is configured properly, and we run the test. The test should restore the state of the Compass Runtime Agent (so it should sync with Compass original configuration).